overlord/snapstate/snapstate.go

```diff
@@ -121,11 +121,23 @@ func doInstall(s *state.State, snapst *SnapState, ss *SnapSetup) (*state.TaskSet
 	linkSnap := s.NewTask("link-snap", fmt.Sprintf(i18n.G("Make snap %q%s available to the system"), ss.Name(), revisionStr))
 	addTask(linkSnap)
 
-	// do GC!
-	if snapst.HasCurrent() {
+	// Do not do that if we are reverting to a local revision
+	if snapst.HasCurrent() && !revisionIsLocal {
```

Comment:
I think this is still what we want (unless I'm missing something of course). I think we should not mess around with garbage-collection or discarding version as long as you jump around local revisions. This code (gc/discard-after-current) will run naturally once the snap is refreshed again to a store version (or when it gets removed).